tests/pytest/testcase.py

```diff
@@ -102,23 +141,52 @@ def get_cli_base_cmd(self) -> str:
             if file.exists():
                 return str(file) + " "
 
-        # this is a little odd... no executable found
+        # this is a little odd... no executable found in "local" directories
         return "cloudtruth "
 
     def get_cmd_env(self):
         return deepcopy(os.environ)
 
+    def get_display_env_command(self) -> str:
+        if os.name == "nt":
+            return "SET"
+        return "printenv"
+
     def run_cli(self, env: Dict[str, str], cmd) -> Result:
+        # WARNING: DOS prompt does not like the single quotes, so use double
+        cmd = cmd.replace("'", "\"")
+
         if self.log_commands:
             print(cmd)
 
+        def _next_part(arg_list: List, key: str) -> str:
+            """Simple function to walk the 'arg_list' and find the item after the 'key'"""
+            for index, value in enumerate(arg_list):
+                if value == key:
+                    return arg_list[index + 1]
+            return None
+
+        # split the command args into something we can work with
+        args = shlex.split(cmd)
+        if "set" in args:
+            # if we're using any of our 'environments' aliases
+            if set(args) & set(["environments", "environment", "envs", "env", "e"]):
+                env_name = _next_part(args, "set")
+                if env_name:
+                    self._environments.add(env_name)
+            # if we're using any of our 'projects' aliases
+            elif set(args) & set(["projects", "project", "proj"]):
+                proj_name = _next_part(args, "set")
+                if proj_name:
+                    self._projects.add(proj_name)
+
         process = subprocess.run(
             cmd, env=env, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE
         )
         result = Result(
             return_value=process.returncode,
-            stdout=process.stdout.decode("utf-8").split("\n"),
-            stderr=process.stderr.decode("utf-8").split("\n"),
+            stdout=process.stdout.decode("us-ascii", errors="ignore").replace("\r", "").split("\n"),
```

Comment:
Are we not testing high-bit characters in names and values?

Reply:
On Windows, we were having problems (presumably with the colors) when dealing with stderr. It was a character mapping issue that can be seen in:
https://github.com/cloudtruth/cloudtruth-cli/runs/2511552892?check_suite_focus=true